src/main/java/workwithJDBC/CustomersDaoImpl.java

```diff
@@ -186,4 +186,45 @@ public boolean deleteCustomer(BigDecimal cust_num) {
         }
         return isRowDeleted;
     }
-}
+
+
+    private abstract class CRUDtemplate <T>{
+
+        public boolean templateOperation(T t) {
+            Connection connection = null;
+            PreparedStatement stmt = null;
+            Boolean isRowDeleted = false;
+            try {
+                connection = Pool.getConnection();
+                stmt = returnPrepareStatement(t, connection);
+                isRowDeleted = stmt.executeUpdate() > 0;
+
+            } catch (SQLException e) {
+                e.printStackTrace();
+            } finally {
+                try {
+                    stmt.close();
+                    connection.close();
+                } catch (SQLException e) {
+                    e.printStackTrace();
+                }
+            }
+            return isRowDeleted;
+        }
+        public abstract PreparedStatement returnPrepareStatement(T t, Connection conn) throws SQLException;
+    }
+
```

Comment:
I think better if this would be in another class.

Reply:
Yes. It critical